WeakAuras/RegionTypes/DynamicGroup.lua

```diff
@@ -48,556 +72,633 @@ function WeakAuras.GetPolarCoordinates(x, y, originX, originY)
+      self.sortedChildren = {}
+      self.controlledChildren = {}
+      self.updatedChildren = {}
+      self.controlPoints:ReleaseAll()
+      for dataIndex, childID in ipairs(data.controlledChildren) do
+        local childRegion, childData = WeakAuras.GetRegion(childID), WeakAuras.GetData(childID)
+        if childRegion and childData then
+          local regionData = createRegionData(childData, childRegion, childID, nil, dataIndex)
+          if childRegion.toShow then
+            tinsert(self.sortedChildren, regionData)
+            self.updatedChildren[regionData] = true
           end
-        else
-          local hiddenXOffset, hiddenYOffset;
-          if(data.grow == "RIGHT") then
-            hiddenXOffset = xOffset - ((childData.width or childRegion.width) + data.space);
-            hiddenYOffset = yOffset - data.stagger;
-          elseif(data.grow == "LEFT") then
-            hiddenXOffset = xOffset + ((childData.width or childRegion.width) + data.space);
-            hiddenYOffset = yOffset - data.stagger;
-          elseif(data.grow == "UP") then
-            hiddenYOffset = yOffset - ((childData.height or childRegion.height) + data.space);
-            hiddenXOffset = xOffset - data.stagger;
-          elseif(data.grow == "DOWN") then
-            hiddenYOffset = yOffset + ((childData.height or childRegion.height) + data.space);
-            hiddenXOffset = xOffset - data.stagger;
-          elseif(data.grow == "CIRCLE" or data.grow == "COUNTERCIRCLE") then
-            hiddenYOffset = cos(angle - angleInc) * radius * -1;
-            hiddenXOffset = sin(angle - angleInc) * radius;
+        end
+        if childData and WeakAuras.clones[childID] then
+          for cloneID, cloneRegion in pairs(WeakAuras.clones[childID]) do
+            local regionData = createRegionData(childData, cloneRegion, childID, cloneID, dataIndex)
+            if cloneRegion.toShow then
+              tinsert(self.sortedChildren, regionData)
+              self.updatedChildren[regionData] = true
+            end
           end
-
-          region.trays[regionData.key]:ClearAllPoints();
-          region.trays[regionData.key]:SetPoint(selfPoint, region, selfPoint, hiddenXOffset, hiddenYOffset);
         end
       end
+      WeakAuras.StopProfileSystem("dynamicgroup")
+      WeakAuras.StopProfileAura(data.id)
+      self:SortUpdatedChildren()
+    else
+      self.needToReload = true
     end
-
-    region:DoResize();
   end
 
-  function region:Suspend()
-    self.suspended = (self.suspended or 0) + 1;
+  function region:AddChild(childID, cloneID)
+    -- adds regionData to the store.
+    -- this is useful mostly for when clones are created which we didn't know about last time Reload was called
+    cloneID = cloneID or ""
+    if self.controlledChildren[childID] and self.controlledChildren[childID][cloneID] then
+      return
+    end
+    local dataIndex
+    for i, id in ipairs(data.controlledChildren) do
+      if id == childID then
```

Comment:
There's actually a function in Blizzard's Util.lua called tIndexOf that does this.

Reply:
thanks